trio/tests/test_util.py

```diff
@@ -58,3 +58,27 @@ def handler(signum, _):
                 async with ul1:
                     pass  # pragma: no cover
     assert "ul1" in str(excinfo.value)
+
+
+async def test_contextmanager_do_not_unchain_non_stopiteration_exceptions():
+    @acontextmanager
+    async def test_issue29692():
+        try:
+            yield
```

Comment:
OK, current version has passing tests for what it's worth, but I'll sign off for now